src/core/time.d

```diff
@@ -2627,103 +2632,82 @@ assert(before + timeElapsed == after).
 
 private:
 
-    // static immutable long _ticksPerSecond;
-
-    unittest
+    static @property long pureTicksPerSecond() @trusted pure nothrow @nogc
     {
-        assert(_ticksPerSecond[_clockIdx]);
+        assert(_ticksPerSecond, "_ticksPerSecond not yet initialized");
+        return _ticksPerSecond;
     }
 
-
-    long _ticks;
-}
-
-// This is supposed to be a static variable in MonoTimeImpl with the static
-// constructor being in there, but https://issues.dlang.org/show_bug.cgi?id=14517
-// prevents that from working. This is the workaround that Steven Schveighoffer
-// came up with. Once issue# 14517 has been fixed, this should be changed so
-// that _ticksPerSecond is a static long in MonoTimeImple rather than a static
-// array outside of it.
-private immutable long[__traits(allMembers, ClockType).length] _ticksPerSecond;
-
-@trusted shared static this()
-{
-    version(Windows)
+    static long getTicksPerSecond() @trusted
     {
-        long ticksPerSecond;
-        if(QueryPerformanceFrequency(&ticksPerSecond) != 0)
+        version(Windows)
         {
-            foreach(i, typeStr; __traits(allMembers, ClockType))
-                _ticksPerSecond[i] = ticksPerSecond;
+            long ticksPerSecond = void;
+            if (QueryPerformanceFrequency(&ticksPerSecond) != 0)
+                return ticksPerSecond;
+            else
+                assert(0, "Failed to get the frequency of the system's monotonic clock.");
         }
-        else
-            assert(0, "Failed to get the frequency of the system's monotonic clock.");
-    }
-    else version(OSX)
-    {
-        mach_timebase_info_data_t info;
-        if(mach_timebase_info(&info) == 0)
+        else version(OSX)
         {
-            long ticksPerSecond = 1_000_000_000L * info.numer / info.denom;
-            foreach(i, typeStr; __traits(allMembers, ClockType))
-                _ticksPerSecond[i] = ticksPerSecond;
+            mach_timebase_info_data_t info = void;
+            if (mach_timebase_info(&info) == 0)
+                return 1_000_000_000L * info.numer / info.denom;
+            else
+                assert(0, "Failed to get the frequency of the system's monotonic clock.");
         }
-        else
-            assert(0, "Failed to get the frequency of the system's monotonic clock.");
-    }
-    else version(Posix)
-    {
-        timespec ts;
-        foreach(i, typeStr; __traits(allMembers, ClockType))
+        else version(Posix)
         {
-            mixin("enum clockType = ClockType." ~ typeStr ~ ";");
-            static if(clockType != ClockType.second)
+            clockid_t clkid = void;
+            version (linux)
```

Comment:
This whole section with `clkid` and the switch statements is unnecessary. `clkid` isn't even used, and because this code is now inside of `MonoTimeImpl` instead of outside of it, it can just use the `clockArg` which is declared at the beginning of `MonoTimeImpl` - and that's what it's currently doing, making this whole section with `clkid` redundant.